messaging/src/test/scala/uk/ac/wellcome/messaging/fixtures/worker/MetricsFixtures.scala

```diff
@@ -1,9 +1,48 @@
 package uk.ac.wellcome.messaging.fixtures.worker
 
+import grizzled.slf4j.Logging
 import org.scalatest.Matchers
+import uk.ac.wellcome.fixtures.TestWith
+import uk.ac.wellcome.messaging.worker.monitoring.MonitoringClient
 
-trait MetricsFixtures extends WorkerFixtures with Matchers {
-  protected def assertMetricCount(metrics: MyMonitoringClient,
+import scala.concurrent.{ExecutionContext, Future}
+
+trait MetricsFixtures extends Matchers {
+
+  class FakeMonitoringClient(shouldFail: Boolean = false)
+    extends MonitoringClient
+      with Logging {
+    var incrementCountCalls: Map[String, Int] = Map.empty
+    var recordValueCalls: Map[String, List[Double]] = Map.empty
+
+    override def incrementCount(metricName: String)(
+      implicit ec: ExecutionContext): Future[Unit] = Future {
+      info(s"MyMonitoringClient incrementing $metricName")
+      if (shouldFail) {
+        throw new RuntimeException(
+          "FakeMonitoringClient incrementCount Error!")
+      }
+      incrementCountCalls = incrementCountCalls + (metricName -> (incrementCountCalls
+        .getOrElse(metricName, 0) + 1))
+    }
+
+    override def recordValue(metricName: String, value: Double)(
+      implicit ec: ExecutionContext): Future[Unit] = Future {
+      info(s"MyMonitoringClient recordValue $metricName: $value")
+      if (shouldFail) {
+        throw new RuntimeException("FakeMonitoringClient recordValue Error!")
+      }
+      recordValueCalls = recordValueCalls + (metricName -> (recordValueCalls
+        .getOrElse(metricName, List.empty) :+ value))
+    }
+  }
+
+  def withFakeMonitoringClient[R](shouldFail: Boolean = false)(testWith: TestWith[FakeMonitoringClient, R]): R = {
+    val fakeMonitoringClient = new FakeMonitoringClient(shouldFail)
+    testWith(fakeMonitoringClient)
+  }
```

Comment:
Again, this feels it would be better suited in the monitoring library.

Reply:
I agree this would be better in the monitoring library, however the `Monitoring` trait depends upon `uk.ac.wellcome.messaging.worker.models.Result`, which is a common messaging concern.  Perhaps we can take a look at this together?